src/SessionSetup/SessionSetupUtils.cpp

```diff
@@ -45,4 +43,28 @@ std::unique_ptr<orbit_client_data::ProcessData> TryToFindProcessData(
   return nullptr;
 }
 
+std::optional<ConnectionTarget> SplitTargetUri(const QString& target_uri) {
+  const QChar kDelimiter = '@';
+
+  QString target_local_copy = target_uri;
+
+  if (!target_local_copy.startsWith(kCustomProtocol + kCustomProtocolDelimiter)) {
+    return std::nullopt;
+  }
+  target_local_copy.remove(0, kCustomProtocol.length() + kCustomProtocolDelimiter.length());
+  if (!target_local_copy.contains(kDelimiter)) {
+    return std::nullopt;
+  }
+
+  const int delimiter_pos = target_local_copy.lastIndexOf('@');
+
+  // There could be more than one delimiter inside the string if the character is allowed
+  // in folder paths - in this case, use the last instance as actual delimiter
+  const QStringRef process_name_or_path = QStringRef(&target_local_copy, 0, delimiter_pos);
+  const QStringRef instance_name_or_id = QStringRef(&target_local_copy, delimiter_pos + 1,
+                                                    target_local_copy.length() - delimiter_pos - 1);
```

Comment:
How about using QUrl for this parsing step? I think it should be able to do all the parsing for you.

Reply:
If you were using the fragment for the process name, then you don't even need to split anything. In that case I would also put the instance id before the process path for hierarchical reasons.

```cpp
QUrl url{target_uri};
if (!url.isValiid() || url.scheme() != kProtocol) return std::nullopt;
return ConnectionTarget{url.filename(), url.authority()};
```

Reply:
(This would also have the advantage that if for whatever reason the URL got encoded with URL encoding or so, this would all be handled by QUrl.)

Reply:
Changed to QUrl, PTAL.